x-pack/plugins/ingest_manager/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -33,31 +37,92 @@ export function getTemplate(
 }
 
 /**
- * Generate mapping takes the given fields array and creates the Elasticsearch
+ * Generate mapping takes the given nested fields array and creates the Elasticsearch
  * mapping properties out of it.
  *
+ * This assumes that all fields with dotted.names have been expanded in a previous step.
+ *
  * @param fields
  */
 export function generateMappings(fields: Field[]): Mappings {
   const props: Properties = {};
   fields.forEach(field => {
-    // Are there more fields inside this field? Build them recursively
-    if (field.fields && field.fields.length > 0) {
-      props[field.name] = generateMappings(field.fields);
-      return;
-    }
-
-    // If not type is defined, take keyword
+    // If type is not defined, assume keyword
     const type = field.type || 'keyword';
-    // Only add keyword fields for now
-    // TODO: add support for other field types
-    if (type === 'keyword') {
-      props[field.name] = { type };
+
+    let fieldProps = getDefaultProperties(field);
+
+    switch (type) {
+      case 'group':
+        fieldProps = generateMappings(field.fields!);
+        break;
+      case 'integer':
+        fieldProps.type = 'long';
+        break;
+      case 'scaled_float':
+        fieldProps.type = 'scaled_float';
+        fieldProps.scaling_factor = field.scaling_factor || DEFAULT_SCALING_FACTOR;
+        break;
+      case 'text':
+        fieldProps.type = 'text';
+        if (field.analyzer) {
+          fieldProps.analyzer = field.analyzer;
+        }
+        if (field.search_analyzer) {
+          fieldProps.search_analyzer = field.search_analyzer;
+        }
+        break;
+      case 'keyword':
+        fieldProps.type = 'keyword';
+        if (field.ignore_above) {
+          fieldProps.ignore_above = field.ignore_above;
+        } else {
+          fieldProps.ignore_above = DEFAULT_IGNORE_ABOVE;
+        }
+        break;
+      // TODO move handling of multi_fields here?
+      case 'object':
+        // TODO improve
+        fieldProps.type = 'object';
+        break;
+      case 'array':
+        // TODO what happens when object_type is not set? beats implementation unclear
```

Comment:
the [unit test](https://github.com/elastic/beats/blob/d9a4c9c240a9820fab15002592e5bb6db318543b/libbeat/template/processor_test.go#L100) says it should return no properties if the object_type is not set.  If there are no properties then I believe [this case](https://github.com/elastic/beats/blob/d9a4c9c240a9820fab15002592e5bb6db318543b/libbeat/template/processor.go#L121) should be met and a key is not generated.

Reply:
I think you're right. The whole mapping generation at this place doesn't account for the case that no field is added at all. I'll add another issue.

Reply:
https://github.com/elastic/kibana/issues/60498

Reply:
... and fixed.